src/main/java/net/neoforged/neoforge/transfer/IResource.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer;
+
+/**
+ * Most general form of a resource that can be quantified and moved around.
+ *
+ * <p>Instances must all be immutable, comparable with {@link Object#equals(Object)}
+ * and they must implement a suitable {@link Object#hashCode()}.
+ */
+public interface IResource {
+    /**
+     * Returns {@code true} if this represents the absence of a resource.
+     *
+     * <p>Examples include item resource with air as an item.
+     */
+    boolean isBlank(); // TODO: potentially use a different name
```

Comment:
`isAbsent()` perhaps to match an existing precedent set by `Optional`.

Reply:
isBlank better than isAbsent

Reply:
isEmpty is better than both but Tech vetoed it

Reply:
I disagree `isBlank` is better, but at this point I say `isEmpty` is the more correct choice. Looking at every data structure I can find `isEmpty` is exactly what the intent is. (I was incorrect on `isAbsent` being an optional method)

Reply:
The reasoning is that a resource being blank doesn't necessarily correspond to a slot or stack being empty.

Reply:
It isn't a matter of if the stack is empty, is if the resource is empty. Which in the case that it matches air, it is

Reply:
Optional<ItemStack>.isEmpty() isn't asking if the itemstack is empty for example, it is asking if the optional contains an itemstack